src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1581,6 +1618,119 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
             }
         }
 
+        /// <summary>
+        /// Validates that deserialization of config file is successful for the following scenarios:
+        /// 1. Nested Mutations section is null
+        /// {
+        ///     "nested-mutations": null
+        /// }
+        /// 
+        /// 2. Nested Mutations section is empty.
+        /// {
+        ///     "nested-mutations": {}
+        /// }
+        ///
+        /// 3. Create field within Nested Mutation section is null.
+        /// {
+        ///     "nested-mutations": {
+        ///         "create": null
+        ///     }
+        /// }
+        ///
+        /// 4. Create field within Nested Mutation section is empty.
+        /// {
+        ///     "nested-mutations": {
+        ///         "create": {}
+        ///     }
+        /// }
+        /// 
+        /// For all the above mentioned scenarios, the expected value for NestedMutationOptions field is null.
+        /// </summary>
+        /// <param name="baseConfig">Base Config Json string.</param>
+        [DataTestMethod]
+        [DataRow(TestHelper.BASE_CONFIG_NULL_NESTED_MUTATIONS_FIELD, DisplayName = "Validate successful deserialization when nested mutation section is null")]
+        [DataRow(TestHelper.BASE_CONFIG_EMPTY_NESTED_MUTATIONS_FIELD, DisplayName = "Validate successful deserialization when nested mutation section is empty")]
+        [DataRow(TestHelper.BASE_CONFIG_NULL_NESTED_CREATE_FIELD, DisplayName = "Validate successful deserialization when create field within nested mutation section is null")]
+        [DataRow(TestHelper.BASE_CONFIG_EMPTY_NESTED_CREATE_FIELD, DisplayName = "Validate successful deserialization when create field within nested mutation section is empty")]
+        public void ValidateDeserializationOfConfigWithNullOrEmptyInvalidNestedMutationSection(string baseConfig)
+        {
+            string configJson = TestHelper.AddPropertiesToJson(baseConfig, BOOK_ENTITY_JSON);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(configJson, out RuntimeConfig deserializedConfig));
+            Assert.IsNotNull(deserializedConfig.Runtime);
+            Assert.IsNotNull(deserializedConfig.Runtime.GraphQL);
+            Assert.IsNull(deserializedConfig.Runtime.GraphQL.NestedMutationOptions);
```

Comment:
is `{ "graphql" : { "nested-mutatations": null } }` valid config?

Reply:
1.  
```json  
  "nested-mutations": null
```

2.
```json
  "nested-mutations": {}
```

3.
```json
  "nested-mutations":{
      "create": { }
  }
```
4.
```json
  "nested-mutations":{
      "create": null 
  }
```


For the above four possibilities, I've taken the following approach:
1. Deserialization of the config file will succeed - there will be no errors (engine would not crash).
2. After deserialization, the `NestedMutationOptions` will be `null`. So, this will result in nested mutation operations being disabled.